src/client/renderingengine.h

```diff
@@ -0,0 +1,175 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2017 nerzhul, Loic Blot <loic.blot@unix-experience.fr>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include "irrlichttypes_extrabloated.h"
+#include "debug.h"
+
+class ITextureSource;
+class Camera;
+class Client;
+class LocalPlayer;
+class Hud;
+class Minimap;
+
+class RenderingEngine
+{
+public:
+	RenderingEngine(IrrlichtDevice *device_);
+	~RenderingEngine();
+
+	v2u32 getWindowSize() const;
+	void setResizable(bool resize);
+
+	video::IVideoDriver *getVideoDriver();
+	static void setXorgClassHint(const video::SExposedVideoData &video_data,
+		const std::string &name);
+	bool setWindowIcon();
+	bool setXorgWindowIconFromPath(const std::string &icon_file);
```

Comment:
setXorgWindowIconFromPath uses Irrlicht:

```
video::IVideoDriver *v_driver = m_device->getVideoDriver();
video::IImageLoader *image_loader = NULL;
```
then it make sense to link it with rendering engine